uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -80,8 +80,8 @@ private static function validateDates($params)
       );
     }
 
-    $startDateIsValid = CRM_HRLeaveAndAbsences_Validator_Date::isValid($params['start_date']);
-    $endDateIsValid = CRM_HRLeaveAndAbsences_Validator_Date::isValid($params['end_date']);
+    $startDateIsValid = CRM_HRLeaveAndAbsences_Validator_Date::isValid($params['start_date'], 'Y-m-d');
+    $endDateIsValid = CRM_HRLeaveAndAbsences_Validator_Date::isValid($params['end_date'], 'Y-m-d');
     if(!$startDateIsValid || !$endDateIsValid) {
```

Comment:
This won't solve the API problem. If we try to create an AbsencePeriod using the API, the validation will fail. I thought the plan was to change the BAO to always expect dates in Ymd format. For that, we would need to:

- Remove the CRM_Utils_Date::processDate call from the create method
- Change the AbsencePeriod form to convert de date to Ymd before passing the form data to the BAO